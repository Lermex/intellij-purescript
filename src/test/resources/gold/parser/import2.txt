Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PSASTWrapperElement(TypeConstructor)
              PsiElement(proper name)('Data')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Management')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(proper name)('Main')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('main')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('x')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(proper name)('User')
          PsiElement(()('(')
          PsiElement(..)('..')
          PsiElement())(')')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(proper name)('Data')
          PsiElement(()('(')
          PsiElement(proper name)('Label')
          PsiElement())(')')
        PsiElement())(')')