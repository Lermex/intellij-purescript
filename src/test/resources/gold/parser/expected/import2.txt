Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSImportDeclarationImpl(IMPORT_DECLARATION)
        PsiElement(PSTokenType.import)('import')
        PSImportNameImpl(IMPORT_NAME)
          PsiElement(PSTokenType.PROPER_NAME)('Data')
          PsiElement(PSTokenType..)('.')
          PSImportNameImpl(IMPORT_NAME)
            PsiElement(PSTokenType.PROPER_NAME)('Management')
            PSDeclarationRefsImpl(DECLARATION_REFS)
              PsiElement(PSTokenType.()('(')
              PSDeclarationRefListImpl(DECLARATION_REF_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('Main')
                PsiElement(PSTokenType.,)(',')
                PSDeclarationRefListImpl(DECLARATION_REF_LIST)
                  PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                    PsiElement(PSTokenType.ID)('main')
                  PsiElement(PSTokenType.,)(',')
                  PSDeclarationRefListImpl(DECLARATION_REF_LIST)
                    PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                      PsiElement(PSTokenType.ID)('x')
                    PsiElement(PSTokenType.,)(',')
                    PSDeclarationRefListImpl(DECLARATION_REF_LIST)
                      PSDeclarationRefTypeConstructorImpl(DECLARATION_REF_TYPE_CONSTRUCTOR)
                        PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                          PsiElement(PSTokenType.PROPER_NAME)('User')
                        PsiElement(PSTokenType.()('(')
                        PsiElement(PSTokenType...)('..')
                        PsiElement(PSTokenType.))(')')
                      PsiElement(PSTokenType.,)(',')
                      PSDeclarationRefListImpl(DECLARATION_REF_LIST)
                        PSDeclarationRefTypeConstructorImpl(DECLARATION_REF_TYPE_CONSTRUCTOR)
                          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                            PsiElement(PSTokenType.PROPER_NAME)('Data')
                          PsiElement(PSTokenType.()('(')
                          PSDeclarationRefTypeConstructorItemImpl(DECLARATION_REF_TYPE_CONSTRUCTOR_ITEM)
                            PsiElement(PSTokenType.PROPER_NAME)('Label')
                          PsiElement(PSTokenType.))(')')
              PsiElement(PSTokenType.))(')')